metrics_reporter.go

```diff
@@ -22,6 +22,17 @@ func sec(t float64) string {
 	return fmt.Sprintf("%.6f", t/1000000000)
 }
 
+type Emitter struct {
+	registry metrics.Registry
+	source   string
+	duration time.Duration
+	logger   *log.Logger
+}
```

Comment:
we should move `lastCounts` thats a package global right now to be a field within `Emitter`. then we can change the signature of `countDifference` to accept `lastCounts` as an argument.

`lastCounts` is the main problem in regard to shared state, as multiple `Emitter`s running at once will wind up clobbering over each other's counts.

Reply:
good call